### PR TITLE
[ansible]: mark variables from custom resources as unsafe

### PR DESCRIPTION
**Pull request by varshaprasad96**
**Description of the change:**
User provided values from the custom resource are tagged
unsafe by default.

**Motivation for the change:**
Closes: #4169 

**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [X] Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))
- [ ] Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)

### CODE REVIEW

**Comment by varshaprasad96** [comment]
cc: @jmrodri @fabianvf

File: internal/ansible/runner/runner.go [hunk 1]
```diff
@@ -337,7 +337,7 @@ func (r *runner) makeParameters(u *unstructured.Unstructured) map[string]interfa
 		parameters = paramconv.MapToSnake(spec)
 	} else {
 		for k, v := range spec {
-			parameters[k] = v
+			parameters[k] = markUnsafe(v)
```
Comment (varshaprasad96) [inline internal/ansible/runner/runner.go:340]: We are also specifying the spec key as it is for json to Marshall (https://github.com/operator-framework/operator-sdk/pull/4566/files#diff-40a9a64aa847bb7816a7eb40cf6ec6b3ca70a7d4066e320d95fe9c0e0cb2ac11R349). Do we need to tag only these variables as unsafe?

Comment (fabianvf) [reply internal/ansible/runner/runner.go:340]: yeah we'll need to run this on all pieces of the spec that we pass in.

File: internal/ansible/runner/runner.go [hunk 2]
```diff
@@ -337,7 +337,7 @@ func (r *runner) makeParameters(u *unstructured.Unstructured) map[string]interfa
 		parameters = paramconv.MapToSnake(spec)
```
Comment (fabianvf) [inline internal/ansible/runner/runner.go:337]: We'd also need to mark these as unsafe

**Review by fabianvf — DISMISSED** [review]
Two more things:
1. Could we also add a setting in the `watches.yaml` to toggle this on/off (off by default is good). As far as I can tell there's not a way to mark a variable as safe once it's been marked unsafe, and some authors may actually want to process user-provided templates (or may be relying on that now)
2. Could we add an integration test with nested values so that we can verify we're properly formatting these variables in a way that Ansible can read

**Review by jmrodri — APPROVED** [review]
/lgtm

**Pull request merged**